src/components/__tests__/Block.test.jsx

```diff
@@ -32,10 +37,69 @@ describe('Components - Block', function() {
   it ('triggers update when its child component changes', function() {
     let block   = app.refine('blocks').first()
     let subject = TestUtils.renderIntoDocument(<Block app={ app } block={ block } />)
+    let params  = { fiz: 'buzz' }
 
-    subject.refs.block.props.onChange({ fiz: 'buzz' })
+    subject.refs.block.props.onChange(params)
 
-    block.content.should.have.property('fiz', 'buzz')
+    app.push.should.have.been.calledWith(Actions.update, block, params)
+  })
+
+  describe('When the "Remove" button is clicked', function() {
+
+    it ('calls the destroy action', function() {
+      let block = app.refine('blocks').first()
+      let test  = TestUtils.renderIntoDocument(<Block app={ app } block={ block } />)
+
+      test.refs.menu.setState({ open: true })
+
+      TestUtils.Simulate.click(test.refs.destroy.getDOMNode())
+
+      app.push.should.have.been.calledWith(Actions.destroy, block.id)
+    })
+  })
+
+  describe('When the "Move Up" button is clicked', function() {
+    it ('calls the move action', function() {
+      let block = app.refine('blocks').last()
+      let test  = TestUtils.renderIntoDocument(<Block app={ app } block={ block } />)
+
+      test.refs.menu.setState({ open: true })
+
+      TestUtils.Simulate.click(test.refs.moveUp.getDOMNode())
+
+      app.push.should.have.been.calledWith(Actions.move, block, -1)
+    })
+
+     it ('is disabled if it is the first block', function() {
+      let block = app.refine('blocks').first()
+      let test = TestUtils.renderIntoDocument(<Block app={ app } block={ block } first />)
+
+      test.refs.menu.setState({ open: true })
+
+      test.refs.moveUp.props.disabled.should.eql(true)
+    })
+  })
+
+  describe('When the "Move Down" button is clicked', function() {
+    it ('calls the move action', function() {
+      let block = app.refine('blocks').first()
+      let test = TestUtils.renderIntoDocument(<Block app={ app } block={ block } />)
+
+      test.refs.menu.setState({ open: true })
+
+      TestUtils.Simulate.click(test.refs.moveDown.getDOMNode())
+
+      app.push.should.have.been.calledWith(Actions.move, block, 1)
+    })
+
+    it ('is disabled if it is the last block', function() {
+      let block = app.refine('blocks').last()
+      let test = TestUtils.renderIntoDocument(<Block app={ app } block={ block } last />)
+
+      test.refs.menu.setState({ open: true })
+
+      test.refs.moveDown.props.disabled.should.eql(true)
+    })
```

Comment:
It should be handled here, actually. I could create a unique test, however it should be covered